array_api_compat/common/_aliases.py

```diff
@@ -54,8 +54,8 @@ def empty_like(
     device: Optional[Device] = None,
     **kwargs,
 ) -> Array:
-    _check_device(xp, device)
-    return xp.empty_like(x, dtype=dtype, **kwargs)
+    with _device_ctx(xp, device, like=x):
```

Comment:
For my education: why can the user pass a `device=` argument to a `*_like` function? Naively I'd have expected that the `_like` implies that the device matches that of `x`. But then you can also pass a `dtype=` which overrides the dtype of `x`, so by symmetry allowing a `device=` makes sense?